iam_role.go

```diff
@@ -17,6 +17,7 @@ type Tag struct {
 type CreateIamRoleOptions struct {
 	RoleName                    *string
 	RoleType                    *string
+	TrustPolicy                 *map[string]interface{}
```

Comment:
Thanks for the callout Webb but we actually do want to validate that the TrustPolicy here is at least valid JSON even if we're deferring the check for whether it's a valid trust policy to ALKS Core. We decided on breaking a little from how AWS handles trust policies since there's a lot of convenience in just passing around JSON and seeing that in the request as JSON rather than a JSON encoded string, so ALKS Core is actually expecting the request body for the create role endpoint to look like
```json
{
  "roleName": "myRole",
  "trustPolicy": {
    "Version": "...",
    
    "Statement": [...]
  },
  ...
}
```
rather than the trustPolicy field being a string containing the JSON, but we do realize that if we decide to change directions on this in the future we can always change the type for TrustPolicy to `interface{}` and then have the CreateIAMRole function detect whether you passed a `string`/`[]byte` or a `map[string]interface{}` and act accordingly so as to preserve backwards compatibility while supporting JSON strings